packages/opentelemetry-resources/src/platform/node/detectors/GcpDetector.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as os from 'os';
+import * as gcpMetadata from 'gcp-metadata';
+import { Resource, Labels } from '../../../Resource';
+import {
+  CLOUD_RESOURCE,
+  HOST_RESOURCE,
+  K8S_RESOURCE,
+  CONTAINER_RESOURCE,
+} from '../../../constants';
+
+/**
+ * The GcpDetector can be used to detect if a process is running in the Google
+ * Cloud Platofrm and return a {@link Resource} populated with metadata about
+ * the instance. Returns an empty Resource if detection fails.
+ */
+export class GcpDetector {
+  static async detect(): Promise<Resource> {
```

Comment:
The AWS detect method has a 2 second maximum, but this one does not. Does gcp-metadata guarantee to block for only a reasonable time?

Reply:
The default timeout for `isAvailable()` is 3000ms, https://github.com/googleapis/gcp-metadata/blob/fbb7158be62c9f1949b69079e35113be1e10495c/src/index.ts#L237